training/src/anemoi/training/train/forecaster.py

```diff
@@ -605,6 +608,26 @@ def calculate_val_metrics(
 
         return metrics
 
+    def on_load_checkpoint(self, checkpoint: dict) -> None:
+        if "rollout" in checkpoint:
+            self.rollout = instantiate(self.config.training.rollout).get_state_from(checkpoint["rollout"])
+
+    def on_save_checkpoint(self, checkpoint: dict) -> None:
+        checkpoint["rollout"] = self.rollout
+
+    def on_train_batch_end(self, *_) -> None:
+        self.rollout.step()
+
+    def on_train_epoch_end(self, *_) -> None:
```

Comment:
Can we add the new rollout step to these logger outputs?

Reply:
As in log the rollout when stepped?

Reply:
I mean adding the rollout to these logging statements:
`LOGGER.debug("Stepping Rollout on train epoch end")`

Reply:
I don't think logging all changes to rollout step in the info terminal output is necessary as the rollout is tracked as a model metric.

Reply:
Resolved in cc060b0db307537ba83ed50fb0d4cab4a0252d79

Reply:
Should the print come after the update of the rollout step? LOGGER.info needs some string formatting.
```suggestion
    def on_train_epoch_end(self, *_) -> None:
        if self.trainer.limit_val_batches == 0:
            self.rollout.step_epoch()
            LOGGER.info("Stepping Rollout on train epoch end, rollout: %i", int(self.rollout))

    def on_validation_epoch_end(self, *_) -> None:
        if not self.trainer.sanity_checking:
            self.rollout.step_epoch()
            LOGGER.info("Stepping Rollout on validation epoch end, rollout: %i", int(self.rollout))
```